### ISSUE TITLE
Missing default margin/padding on direction: right/left with 3 embedded shapes? 

### ISSUE THREAD

**Issue by czardien** [terrastruct/d2#968]
Hi, 

I'm seeing a slightly different behaviour in the margin/padding between directions `up/down` and `left/right`, where a margin/padding will exist in the former but not in the latter, when have more than 2 embedded shapes.

To see the behaviour, one can compare the output of the following script:

```
direction: up

A
B
B.C

A.a
B.C.b

A.a -> B.C.b
```

... and:

```
direction: right

A
B
B.C

A.a
B.C.b

A.a -> B.C.b
```

... see attached for the results, which are interchangeable for `{up|down}` and `{left|right}`. 

We can clearly see an "expected" margin/padding in the former script, which we can't see in the latter.

![right-no-margin-padding](https://user-images.githubusercontent.com/19569946/222760131-64e143c6-1041-47be-a8f0-1a38dc6af941.svg)
![up](https://user-images.githubusercontent.com/19569946/222760140-c0fae1d0-901b-4846-95e0-47a0e546dcd8.svg)

**Comment by alixander** [comment]
yes i think related to https://github.com/terrastruct/d2/issues/925

we'll get this fixed this release (probably release tonight). thank you for reporting.

**Comment by czardien** [comment]
I should mention I'm running `d2` version `0.2.2` which I installed from the [Arch package](https://archlinux.org/packages/community/x86_64/d2/).

```
$ d2 -v
0.2.2
```

**Comment by gavin-ts** [comment]
https://play.d2lang.com/?script=goe5lUIRKGC4uBy5nLic9Jy5uBz1EkEMvSQwU0HXTgHCAwQAAP__&

![Screen Shot 2023-03-14 at 12 51 43 PM](https://user-images.githubusercontent.com/85081687/225120888-c6332524-e8e8-4892-8edd-ba3a534589ce.png)

I think this was fixed in #973
